zaza/openstack/charm_tests/ceph/rbd_mirror/tests.py

```diff
@@ -197,9 +204,12 @@ def get_pools(self):
                 'ceph-mon' + self.site_b_app_suffix,
                 model_name=self.site_b_model),
             model_name=self.site_b_model)
+        if not include_internal_pools:
+            remove_internal_pools(site_a_pools)
+            remove_internal_pools(site_b_pools)
         return sorted(site_a_pools.keys()), sorted(site_b_pools.keys())
```

Comment:
Having just indicated above in `remove_internal_pools` that the function should be immutable, it occurs to me here, that this could be better written as:

```python
pools = set(itertools.chain(*site_a_pools,.keys(), *site_b_pools.keys())
return sorted(pools.difference(INTERNAL_POOLS))
```

Note, sorted returns an iterator, so the :rtype: on the function is wrong anyway.  It would need `tuple(..)` or `list(..)` to force a data structure return